test/node-fetch/main.js

```diff
@@ -1347,42 +1347,42 @@ describe('node-fetch', () => {
     })
   })
 
-  it('should allow cloning a json response and log it as text response', () => {
-    const url = `${base}json`
-    return fetch(url).then(res => {
-      const r1 = res.clone()
-      return Promise.all([res.json(), r1.text()]).then(results => {
-        expect(results[0]).to.deep.equal({ name: 'value' })
-        expect(results[1]).to.equal('{"name":"value"}')
-      })
-    })
-  })
-
-  it('should allow cloning a json response, and then log it as text response', () => {
-    const url = `${base}json`
-    return fetch(url).then(res => {
-      const r1 = res.clone()
-      return res.json().then(result => {
-        expect(result).to.deep.equal({ name: 'value' })
-        return r1.text().then(result => {
-          expect(result).to.equal('{"name":"value"}')
-        })
-      })
-    })
-  })
-
-  it('should allow cloning a json response, first log as text response, then return json object', () => {
-    const url = `${base}json`
-    return fetch(url).then(res => {
-      const r1 = res.clone()
-      return r1.text().then(result => {
-        expect(result).to.equal('{"name":"value"}')
-        return res.json().then(result => {
-          expect(result).to.deep.equal({ name: 'value' })
-        })
-      })
-    })
-  })
+  // it('should allow cloning a json response and log it as text response', () => {
```

Comment:
@KhafraDev

Reply:
stream is being locked somewhere

Reply:
Specifically just calling `response.clone()` will throw an error. Happens in `cloneBody`'s first step (teeing the stream). No idea how the stream is locked beforehand as it shouldn't be used.

Reply:
```diff
diff --git a/lib/fetch/response.js b/lib/fetch/response.js
index 2e7e42e..53d9794 100644
--- a/lib/fetch/response.js
+++ b/lib/fetch/response.js
@@ -316,12 +316,6 @@ function cloneResponse (response) {
   // 1. If response is a filtered response, then return a new identical
   // filtered response whose internal response is a clone of response’s
   // internal response.
-  if (response.internalResponse) {
-    return filterResponse(
-      cloneResponse(response.internalResponse),
-      response.type
-    )
-  }

   // 2. Let newResponse be a copy of response, except for its body.
   const newResponse = makeResponse({ ...response, body: null })

```

`response.body` and `response.internalResponse.body` are mismatched; response's body's stream is not locked and works properly (yay!), but internalResponse's body's stream is locked, causing the error. The above is NOT a good solution but all tests do pass...

Reply:
```diff
diff --git a/lib/fetch/response.js b/lib/fetch/response.js
index 2e7e42e..9756273 100644
--- a/lib/fetch/response.js
+++ b/lib/fetch/response.js
@@ -316,11 +316,11 @@ function cloneResponse (response) {
   // 1. If response is a filtered response, then return a new identical
   // filtered response whose internal response is a clone of response’s
   // internal response.
-  if (response.internalResponse) {
-    return filterResponse(
-      cloneResponse(response.internalResponse),
-      response.type
-    )
+  if (response.internalResponse != null) {
+    return filterResponse({
+      ...response.internalResponse,
+      body: cloneBody(response.body)
+    }, response.type)
```
This also works but I'm just throwing stuff against the wall and hoping it sticks tbh.

Reply:
I feel like we are miss-reading the spec somewhere somehow...

Reply:
I think I figured it out. A filtereredResponse should be a proxy for the response.